File: onething/boot/api/src/main/java/com/clip/api/user/controller/dto/SignupDto.java
package com.clip.api.user.controller.dto;

import com.clip.user.entity.Platform;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@NoArgsConstructor
public class SignupDto {
    private boolean servicePermission;
    private boolean privatePermission;
    private boolean marketingPermission;
    private String socialId;
    private Platform platform;

    @Builder
    public SignupDto(boolean servicePermission, boolean privatePermission, boolean marketingPermission, String socialId, Platform platform) {
        this.servicePermission = servicePermission;
        this.privatePermission = privatePermission;
        this.marketingPermission = marketingPermission;
        this.socialId = socialId;
        this.platform = platform;
    }
}

File: onething/adapter/api/src/main/kotlin/com/clip/adapter/primary/web/user/dto/SignUpDTO.kt
package com.clip.adapter.primary.web.user.dto

import com.clip.domain.user.User
import com.clip.domain.user.enums.DeviceType
import com.clip.domain.user.enums.Platform
import jakarta.validation.constraints.AssertTrue
import jakarta.validation.constraints.NotBlank

data class SignUpDTO(
    @field:AssertTrue(message = "서비스 이용약관에 동의해야 합니다.")
    val servicePermission: Boolean,
    @field:AssertTrue(message = "개인정보 수집 및 이용에 동의해야 합니다.")
    val privatePermission: Boolean,
    val marketingPermission: Boolean,
    val platform: Platform,
    @field:NotBlank
    val socialId: String,
    @field:NotBlank
    val osVersion: String,
    val deviceType: DeviceType,
    val isAllowNotify: Boolean,
    val firebaseToken: String?,
){
    fun toUser(): User {
        return User.register(
            servicePermission = servicePermission,
            privatePermission = privatePermission,
            marketingPermission = marketingPermission,
            platform = platform,
            socialId = socialId,
            deviceType = deviceType,
            osVersion = osVersion,
            isAllowNotify = isAllowNotify,
            firebaseToken = firebaseToken
        )
    }
}
